metals/src/main/scala/scala/meta/internal/parsing/BaseToken.scala

```diff
@@ -0,0 +1,50 @@
+package scala.meta.internal.parsing
+
+import scala.meta.inputs.Input
+import scala.meta.inputs.Position
+import scala.meta.tokens.Token
+
+sealed trait BaseToken {
+  def pos: Position
+  def input: Input
+  def structure: String
+  def start: Int
+  def end: Int
+  def isLF: Boolean
+}
+
+case class ScalaToken(token: Token) extends BaseToken {
+
+  override def pos: Position = token.pos
```

Comment:
You could use `val` instead of def in overrides

Reply:
This is no longer needed, so removed it, but good idea. I used it in the JavaToken